prover/src/provers/prove_prover.rs

```diff
@@ -0,0 +1,83 @@
+use super::Prover;
+use crate::contexts::ProveContext;
+
+use num::ToPrimitive;
+use serde::Serialize;
+use std::fs;
+use std::time::Duration;
+
+use plonky2::field::goldilocks_field::GoldilocksField;
+use plonky2::plonk::config::{GenericConfig, PoseidonGoldilocksConfig};
+use plonky2::util::timing::TimingTree;
+
+// use plonky2x::backend::wrapper::wrap::WrappedCircuit;
+// use plonky2x::frontend::builder::CircuitBuilder as WrapperBuilder;
+
+
+use zkm::all_stark::AllStark;
+use zkm::config::StarkConfig;
+use zkm::cpu::kernel::assembler::segment_kernel;
+use zkm::fixed_recursive_verifier::AllRecursiveCircuits;
+use zkm::mips_emulator::state::{InstrumentedState, State, SEGMENT_STEPS};
+
+use std::fs::File;  
+use std::io::Write;
+
+#[derive(Default)]
+pub struct ProveProver {}
+
+impl ProveProver {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Prover<ProveContext> for ProveProver {
```

Comment:
ProveProver can be renamed to RootProver

Reply:
done